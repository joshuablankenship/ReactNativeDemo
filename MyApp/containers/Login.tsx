import React, {Component} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {styles} from '../styles/LoginStyles';
import {authorize} from 'react-native-app-auth';
import config from '../services/AuthService';
interface Props {
  navigation: any;
}
export default class Login extends Component<Props> {
  _onPressButton = async () => {
    // use the client to make the auth request and receive the authState
    try {
      // const result = await authorize(config);
      this.props.navigation.navigate('Main');
      // console.warn(result);
      // result includes accessToken, accessTokenExpirationDate and refreshToken
    } catch (error) {
      console.log(error);
    }
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.container}>
          <Image source={require('../images/PYh.gif')} />
          <View style={styles.alternativeLayoutButtonContainer} />
          <TouchableOpacity style={styles.button} onPress={this._onPressButton}>
            <Image source={require('../images/Google.png')} />
            <Text> Login with Google</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
