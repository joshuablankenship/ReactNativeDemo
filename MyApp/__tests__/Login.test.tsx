// eslint-disable-next-line @typescript-eslint/no-unused-vars
import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {TouchableOpacity} from 'react-native';
import Login from '../containers/Login';

const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  increment: jest.fn(),
  ...props,
});

describe('Test Login Screen', () => {
  let wrapper: ShallowWrapper;
  let props: any;
  it('It should match initial snapshot', () => {
    props = createTestProps({
      counter: {
        value: 0,
        isUpdating: false,
      },
    });
    wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('It should display Login with Google button', () => {
    props = createTestProps;
    wrapper = shallow(<Login {...props} />);
    expect(wrapper.find(TouchableOpacity).render().text()).toBe(
      ' Login with Google',
    );
  });
});
