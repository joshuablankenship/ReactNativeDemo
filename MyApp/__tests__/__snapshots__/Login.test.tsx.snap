// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Screen It should match initial snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/PYh.gif",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 20,
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#859a9b",
          "borderRadius": 20,
          "flex": -1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginLeft": 110,
          "marginRight": 110,
          "padding": 10,
          "shadowColor": "#303838",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/Google.png",
          }
        }
      />
      <Text>
         Login with Google
      </Text>
    </ForwardRef>
  </View>
</View>
`;
